### ISSUE TITLE
Multiprocessing error in check_dl1

### ISSUE THREAD

**Issue by morcuended** [cta-observatory/cta-lstchain#416]
Here is another error I've been getting during the reprocessing. In this case, it seems to be related to the multiprocessing part in the `check_dl1` function included in the r0_to_dl1 step. When this error raises, datacheck h5 file is not produced whereas the DL1 and muon files are normally generated.

```python
multiprocessing.pool.RemoteTraceback:
"""
Traceback (most recent call last):
  File "/fefs/aswg/software/virtual_env/anaconda3/envs/osa/lib/python3.7/site-packages/numpy/lib/histograms.py", line 1018, in histogramdd
    N, D = sample.shape
AttributeError: 'list' object has no attribute 'shape'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/fefs/aswg/software/virtual_env/anaconda3/envs/osa/lib/python3.7/multiprocessing/pool.py", line 121, in worker
    result = (True, func(*args, **kwds))
  File "/fefs/aswg/software/virtual_env/anaconda3/envs/osa/lib/python3.7/multiprocessing/pool.py", line 47, in starmapstar
    return list(itertools.starmap(args[0], args[1]))
  File "/fefs/aswg/software/virtual_env/anaconda3/envs/osa/lib/python3.7/site-packages/lstchain/datachecks/dl1_checker.py", line 280, in process_dl1_file
    geom, bins)
  File "/fefs/aswg/software/virtual_env/anaconda3/envs/osa/lib/python3.7/site-packages/lstchain/datachecks/containers.py", line 205, in fill_event_wise_info
    bins=histogram_binnings.
  File "/fefs/aswg/software/virtual_env/anaconda3/envs/osa/lib/python3.7/site-packages/matplotlib/pyplot.py", line 2621, in hist2d
    data is not None else {}), **kwargs)
  File "/fefs/aswg/software/virtual_env/anaconda3/envs/osa/lib/python3.7/site-packages/matplotlib/__init__.py", line 1565, in inner
    return func(ax, *map(sanitize_sequence, args), **kwargs)
  File "/fefs/aswg/software/virtual_env/anaconda3/envs/osa/lib/python3.7/site-packages/matplotlib/cbook/deprecation.py", line 296, in wrapper
    return func(*args, **kwargs)
  File "/fefs/aswg/software/virtual_env/anaconda3/envs/osa/lib/python3.7/site-packages/matplotlib/axes/_axes.py", line 6915, in hist2d
    normed=density, weights=weights)
  File "<__array_function__ internals>", line 6, in histogram2d
  File "/fefs/aswg/software/virtual_env/anaconda3/envs/osa/lib/python3.7/site-packages/numpy/lib/twodim_base.py", line 715, in histogram2d
    hist, edges = histogramdd([x, y], bins, range, normed, weights, density)
  File "<__array_function__ internals>", line 6, in histogramdd
  File "/fefs/aswg/software/virtual_env/anaconda3/envs/osa/lib/python3.7/site-packages/numpy/lib/histograms.py", line 1021, in histogramdd
    sample = np.atleast_2d(sample).T
  File "<__array_function__ internals>", line 6, in atleast_2d
  File "/fefs/aswg/software/virtual_env/anaconda3/envs/osa/lib/python3.7/site-packages/numpy/core/shape_base.py", line 123, in atleast_2d
    ary = asanyarray(ary)
  File "/fefs/aswg/software/virtual_env/anaconda3/envs/osa/lib/python3.7/site-packages/numpy/core/_asarray.py", line 138, in asanyarray
    return array(a, dtype, copy=False, order=order, subok=True)
  File "/fefs/aswg/software/virtual_env/anaconda3/envs/osa/lib/python3.7/site-packages/pandas/core/series.py", line 871, in __getitem__
    result = self.index.get_value(self, key)
  File "/fefs/aswg/software/virtual_env/anaconda3/envs/osa/lib/python3.7/site-packages/pandas/core/indexes/base.py", line 4404, in get_value
    return self._engine.get_value(s, k, tz=getattr(series.dtype, "tz", None))
  File "pandas/_libs/index.pyx", line 80, in pandas._libs.index.IndexEngine.get_value
  File "pandas/_libs/index.pyx", line 90, in pandas._libs.index.IndexEngine.get_value
  File "pandas/_libs/index.pyx", line 138, in pandas._libs.index.IndexEngine.get_loc
  File "pandas/_libs/hashtable_class_helper.pxi", line 998, in pandas._libs.hashtable.Int64HashTable.get_item
  File "pandas/_libs/hashtable_class_helper.pxi", line 1005, in pandas._libs.hashtable.Int64HashTable.get_item
KeyError: 0
"""
The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/fefs/aswg/software/virtual_env/anaconda3/envs/osa/bin/lstchain_data_r0_to_dl1", line 8, in <module>
    sys.exit(main())
  File "/fefs/aswg/software/virtual_env/anaconda3/envs/osa/lib/python3.7/site-packages/lstchain/scripts/lstchain_data_r0_to_dl1.py", line 163, in main
    tib_counter0=args.tib_counter0
  File "/fefs/aswg/software/virtual_env/anaconda3/envs/osa/lib/python3.7/site-packages/lstchain/reco/r0_to_dl1.py", line 676, in r0_to_dl1
    max_cores=1, create_pdf=False)
  File "/fefs/aswg/software/virtual_env/anaconda3/envs/osa/lib/python3.7/site-packages/lstchain/datachecks/dl1_checker.py", line 126, in check_dl1
    dl1datacheck = pool.starmap(process_dl1_file, func_args)
  File "/fefs/aswg/software/virtual_env/anaconda3/envs/osa/lib/python3.7/multiprocessing/pool.py", line 276, in starmap
    return self._map_async(func, iterable, starmapstar, chunksize).get()
  File "/fefs/aswg/software/virtual_env/anaconda3/envs/osa/lib/python3.7/multiprocessing/pool.py", line 657, in get
    raise self._value
KeyError: 0
```

**Comment by moralejo** [comment]
Thanks for reporting. I am looking into it. I do not think it is really related to multiprocessing (actually only one process is run since each subrun is processed separately), it is related to some histogramming, apparently it gets a list when it is expecting and ndarray.  Does that ring a bell to anyone?

**Comment by moralejo** [comment]
Should be solved by #417

**Comment by rlopezcoto** [comment]
Solved via #417
